TINY-6587: Review Autocompleter, Split Button, and Menu Button page (#2000)
tinymce/tinymce-docs: ui-components/autocompleter.md

## Overview

> An autocompleter displays suggestions while the user is typing. Suggestions are shown when the trigger character is typed in (e.g. ':'). Pressing `esc` should close the autocompleter.

Reviewer: ```suggestion
An autocompleter displays suggestions while the user is typing. Suggestions are shown when the trigger character is typed in (such as '`:`'). Pressing the Escape key should close the autocompleter.
```

Reviewer: FYI: https://www.tiny.cloud/docs/configure/contributing-docs/#keepitsimple re `e.g.`

> An autocompleter displays suggestions while the user is typing. Suggestions are shown when the trigger character is typed in (such as '`:`'). Pressing the Escape key should close the autocompleter.

Reviewer 2: We've lost this bit of data in this rewrite, can you please make sure that remains as it's important:
> the user must precede the trigger character (colon in this case) with a separator character, such as a space or a newline.

Reviewer: ```suggestion
An autocompleter displays suggestions while the user is typing. Suggestions are shown when the trigger character is entered after a space or at the start of a new line (such as '` :`'). Pressing the Escape key should close the autocompleter.
```

> An autocompleter displays suggestions while the user is typing. Suggestions are shown when the trigger character is typed in (such as '`:`'). Pressing the Escape key should close the autocompleter.

Reviewer 3: `should close` or `will close`?

Reviewer: ```suggestion
An autocompleter displays suggestions while the user is typing. Suggestions are shown when the trigger character is typed in (such as '`:`'). Pressing the Escape key will close the autocompleter.
```

Author: I've changed to `will close`.

> ## Interactive example

Reviewer: ```suggestion
## Interactive examples
```

> The following shows two examples of how to create a charmap autocompleter. The first example is constructed using the standard autocompleter item and will show whenever a `:` character is typed plus at least one additional character. The second is constructed using [CardMenuItems](#cardmenuitem) and will show whenever a `-` character is typed plus at least one additional character.

Reviewer: ```suggestion
The following examples show how to create a special characters autocompleter. The first example uses the standard autocompleter item and will show when user types the colon (`:`) character and at least one additional character. The second uses [CardMenuItems](#cardmenuitem) and will show when a user types a hyphen (`-`) character and at least one additional character.
```
